webui/react/src/pages/Cluster/ClusterHistoricalUsageChart.tsx

```diff
@@ -87,30 +80,31 @@ const ClusterHistoricalUsageChart: React.FC<ClusterHistoricalUsageChartProps> =
       height,
       series,
       tzDate: ts => uPlot.tzDate(new Date(ts * 1e3), 'Etc/UTC'),
-      width: chartRef.current.offsetWidth,
-    } as Options;
-
-    const plotChart = new uPlot(options, data as uPlot.AlignedData, chartRef.current);
-    setChart(plotChart);
-
-    return () => {
-      setChart(undefined);
-      plotChart.destroy();
-    };
+    });
   }, [ groupBy, hasData, height, hoursByLabel, hoursTotal, time ]);
 
-  // Resize the chart when resize events happen.
   useEffect(() => {
-    if (chart) chart.setSize({ height, width: resize.width });
-  }, [ chart, height, resize ]);
+    const timeUnix: number[] = time.map(item => Date.parse(item) / 1000);
+
+    const data: AlignedData = [ timeUnix ];
+    if (hoursTotal) {
+      data.push(hoursTotal);
+    }
+
+    Object.keys(hoursByLabel).forEach(label => {
+      data.push(hoursByLabel[label]);
+    });
+
+    setChartData(data);
+  }, [ hoursByLabel, hoursTotal, time ]);
```

Comment:
**suggestion:** this could be a `useMemo` instead of a `useEffect` + `useState`.

Reply:
great, great, great suggestion, thanks!